pkg/runner/cluster_swarm.go

```diff
@@ -348,3 +354,19 @@ func (*ClusterSwarmRunner) ConfigType() reflect.Type {
 func (*ClusterSwarmRunner) CompatibleBuilders() []string {
 	return []string{"docker:go"}
 }
+
+func retry(attempts int, sleep time.Duration, f func() error) (err error) {
+	for i := 0; ; i++ {
+		err = f()
+		if err == nil {
+			return
+		}
+
+		if i >= (attempts - 1) {
+			break
+		}
+
+		time.Sleep(sleep)
+	}
+	return fmt.Errorf("after %d attempts, last error: %s", attempts, err)
```

Comment:
```suggestion
	return fmt.Errorf("failed after %d attempts, last error: %w", attempts, err)
```

Reply:
(to wrap the error, thanks go1.13)

Reply:
Agree, it would be good to adopt the new errors package, but we wrap errors at many places, so will leave this for a future PR.

Reply:
No need to adopt the errors package, it's part of `fmt.Errorf()` now: https://golang.org/pkg/fmt/#Errorf